transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/SourceSetUtils.java

```diff
@@ -0,0 +1,101 @@
+import org.gradle.api.plugins.Convention;
+import org.gradle.api.tasks.ScalaSourceSet;
+import org.gradle.api.tasks.SourceSet;
+
+
+/**
+ * Utility class to help manipulate a {@link SourceSet}
+ */
+class SourceSetUtils {
+
+  private SourceSetUtils() {
+  }
+
+  /**
+   * Returns the {@link SourceDirectorySet} for a given {@link SourceSet} depending on the language of the sources
+   */
+  static SourceDirectorySet getSourceDirectorySet(SourceSet sourceSet, Language language) {
+    switch (language) {
+      case JAVA:
+        return sourceSet.getJava();
+      case SCALA:
+        Convention sourceSetConvention = (Convention) InvokerHelper.getProperty(sourceSet, "convention");
+        ScalaSourceSet scalaSourceSet = sourceSetConvention.getPlugin(ScalaSourceSet.class);
+        return scalaSourceSet.getScala();
+      default:
+        throw new UnsupportedOperationException("Language " + language + " not supported");
+    }
+  }
+
+  private static Configuration getConfigurationForSourceSet(Project project, SourceSet sourceSet,
+      DependencyConfigurationType configurationName) {
+    return project.getConfigurations().getByName(getConfigurationNameForSourceSet(sourceSet, configurationName));
+  }
+
+  private static String getConfigurationNameForSourceSet(SourceSet sourceSet,
+      DependencyConfigurationType configurationName) {
+    final String configName;
+    switch (configurationName) {
+      case ANNOTATION_PROCESSOR:
+        configName = sourceSet.getAnnotationProcessorConfigurationName();
+        break;
+      case IMPLEMENTATION:
+        configName = sourceSet.getImplementationConfigurationName();
+        break;
+      case COMPILE_ONLY:
+        configName = sourceSet.getCompileOnlyConfigurationName();
+        break;
+      case RUNTIME_ONLY:
+        configName = sourceSet.getRuntimeOnlyConfigurationName();
+        break;
+      default:
+        throw new UnsupportedOperationException("Configuration " + configurationName + " not supported");
+    }
+    return configName;
+  }
+
+  /**
+   * Adds the provided dependency to the given {@link Configuration}
+   */
+  static void addDependencyToConfiguration(Project project, Configuration configuration, Object dependency) {
+    configuration.withDependencies(dependencySet -> dependencySet.add(project.getDependencies().create(dependency)));
+  }
+
+  /**
+   * Adds the provided dependencies to the given {@link Configuration}
+   */
+  static void addDependenciesToConfiguration(Project project, Configuration configuration,
+      Collection<DependencyConfiguration> dependencies) {
+    dependencies.forEach(
+        dependency -> addDependencyToConfiguration(project, configuration, dependency.getDependencyString()));
+  }
+
+  /**
+   * Adds the provided dependency to the appropriate configurations of the given {@link SourceSet}
+   */
+  static void addDependencyToSourceSet(Project project, SourceSet sourceSet, DependencyConfiguration dependency) {
```

Comment:
dependency --> dependencyConfiguration here and everywhere

Reply:
Actually seems the class name should be dependency. Let me know what you think.

Reply:
I tried using a Multimap for ConfigurationDependencies as we discusses offline. This was making `Defaults.java` much more complicated without significant benefits IMO. I keep the `DependencyConfiguration` class but make the variable naming consistent.